node/src/reactor/joiner.rs

```diff
@@ -526,20 +540,22 @@ impl reactor::Reactor for Reactor {
         let chainspec = chainspec_loader.chainspec().clone();
         let linear_chain_sync = match maybe_trusted_hash {
             None => {
-                if let Some(start_time) = chainspec
+                let genesis_timestamp = chainspec
                     .protocol_config
                     .activation_point
-                    .genesis_timestamp()
-                {
-                    let era_duration = chainspec.core_config.era_duration;
-                    if Timestamp::now() > start_time + era_duration {
-                        error!(
+                    .genesis_timestamp();
+                let era_duration = chainspec.core_config.era_duration;
+                let genesis_era_ended = genesis_timestamp.map_or(true, |start_time| {
+                    Timestamp::now() > start_time + era_duration
```

Comment:
(Not really new in this PR, but:)
Shouldn't we allow omitting the trusted hash until `start_time + unbonding_period`?

Reply:
I think that should be okay in theory. Although do we still keep this many eras? I seem to remember that we started removing eras from memory earlier at some point, but maybe it still wasn't earlier than the unbonding period? I can't recall the details.

Reply:
How many eras we keep in memory in consensus shouldn't matter:
This is about whether the finality signatures of the genesis validators can still be trusted.

Reply:
I'll leave this for another time, if that's okay.